src/calculators/BarnesPointsCalc.ts

```diff
@@ -0,0 +1,134 @@
+import { Results } from "../common/CrewTimerTypes";
+import {
+  genPlaces,
+  isAFinal,
+} from "../common/CrewTimerUtils";
+
+export type BarnesPointsResults = {
+  team: string;
+  points: number;
+  place: number;
+}[];
+
+export const maxPointsFromName = (eventName: string) => {
+  eventName = eventName.toUpperCase();
+  const boatClassCaptureExpression = /.* ([1248]+)[x\-\+]+/;
+  const match = eventName.match(boatClassCaptureExpression);
+
+  if (match && match.length > 1) {
+    const boatClass = Number.parseInt(match[1]);
+    if (boatClass == 1) {
+      return 10;
+    }
+    if (boatClass == 2) {
+      return 15;
+    }
+    if (boatClass == 4) {
+      return 20;
+    }
+    if (boatClass == 8) {
+      return 30;
+    }
+  }
+
+  return 0;
```

Comment:
I have a similar function in CrewTimerUtils that returns the number of seats - numSeatsFromName().  In mine I return 1 if the boat class is not recognized on the assumption there is at least one rower on board.